fix(minimongo): `update/remove` with operator only applies one record

ggazzo opened the pull request:
<!--
First, 🌠 thank you 🌠 for taking the time to consider a contribution to Meteor!

Here are some important details to follow:

* ⏰ Your time is important
          To save your precious time, if the contribution you are making will take more
          than an hour, please make sure it has been discussed in an issue first.
          This is especially true for feature requests!
* 💡 Features
          Feature requests can be created and discussed by visiting:
          https://github.com/meteor/meteor/discussions
* 🕷 Bug fixes
          These can be created and discussed in this repository. When fixing a bug,
          please _try_ to add a test which verifies the fix.  If you cannot, you should
          still submit the PR but we may still ask you (and help you!) to create a test.
* 📖 Contribution guidelines
          Always follow https://github.com/meteor/meteor/blob/master/CONTRIBUTING.md
          when submitting a pull request.  Make sure existing tests still pass, and add
          tests for all new behavior.
* ✏️ Explain your pull request
          Describe the big picture of your changes here to communicate to what your
          pull request is meant to accomplish.  Provide 🔗 links 🔗 to associated issues!

We hope you will find this to be a positive experience!  Open source contribution can be intimidating and we hope to alleviate that pain as much as possible.  Without following these guidelines, you may be missing context that can help you succeed with your contribution, which is why we encourage discussion first.  Ultimately, there is no guarantee that we will be able to merge your pull-request, but by following these guidelines we can try to avoid disappointment.
-->


checking the code we can see that in the past we were used to do `fn(doc, id) === false`
in this pr https://github.com/meteor/meteor/pull/12504 was changed to `!fn(doc, id)`

`fn(doc, id) === false` and `!fn(doc, id)` are slightly different but causes huge problem if we are considering/ignoring undefined

if we use operators like `$in`, `uptade`/`remove` uses  `_eachPossiblyMatchingDocSync` which returns `undefined`

```diff
if (specificIds) {
-      specificIds.some(id => {
-        const doc = this._docs.get(id);
-        if (doc) {
-          return fn(doc, id) === false;
-        }
-       });
+      for (const id of specificIds) {
+        const doc = this._docs.get(id);
+        if (doc && !fn(doc, id)) {
+          break
+        }
+      }
}
```

with that only one record is affected by the operation

graemian committed:
Add instructions for enabling back gesture on iOS

ebroder committed:
Re-enable TypeScript self-test

The latest release of zodern:types is able to pull type declarations
from a Meteor checkout, if it's being used, instead of only from the
package repository (which may not contain the latest changes). This
means that we should be able to re-enable the TypeScript test, which
verifies that the TypeScript skeleton does, in fact, produce a project
which passes type checking.

(Re-enabling the test also picked up a few latent bugs in the type
declarations, so fix those as well)

zodern committed:
Add missing dependency on jquery

zodern committed:
Fix error generating xunit output

Grubba27 committed:
Merge branch 'devel' into fix/test-in-console

Grubba27 committed:
to version 1.2.7

StorytellerCZ committed:
Merge branch 'devel' into fix/test-in-console

nachocodoner committed:
Merge branch 'devel' into fix/test-in-console

mjcctech committed:
fix: displaying google maps in mobile application

dupontbertrand committed:
Update docs for upsert in ServiceConfiguration

Commit:
Update 3.forms-and-events.md

In the instructions to add onSubmit and onChange to TaskForm.jsx there was an unused import for TaskCollections. Removing this had no effect on the useability of the code in this step.

ggazzo committed:
fix: minimongo $in only applies one record

ggazzo committed:
adds tests to update method

ggazzo commented:
@denihs do you need me to do anything else here?

ggazzo committed:
Merge branch 'devel' into fix/minimongo-remove-in

fredmaiaarantes committed:
Fix CLI and ecmascript links and add leonardoventurini:scss as the recommended scss package.

ggazzo committed:
Merge branch 'devel' into fix/minimongo-remove-in

ggazzo committed:
Merge branch 'devel' into fix/minimongo-remove-in

ggazzo committed:
Merge branch 'devel' into fix/minimongo-remove-in

ggazzo committed:
Merge branch 'devel' into fix/minimongo-remove-in

Grubba27 commented:
`tinytest - minimongo - updateAsync` multiple using query operator failed seems to be failing

carlosarraes committed:
feat: add support for dots in minimongo key names

Minimongo now allows dots (.) in MongoDB key names, aligning with modern
MongoDB support.

Fixes #13552

nachocodoner committed:
update caching strategy in meteor-selftest-windows workflow

nachocodoner committed:
use Windows tar instead of Git tar in workflow

italojs committed:
Merge branch 'devel' into fix-link-and-recommended-scss

italojs committed:
Merge branch 'devel' into patch-1

italojs committed:
Merge pull request #13428 from JacobKGilbert/patch-1

Update 3.forms-and-events.md

italojs committed:
Merge branch 'devel' into patch-2

italojs committed:
Merge pull request #13404 from dupontbertrand/patch-2

Update docs for upsert in ServiceConfiguration

italojs committed:
Merge branch 'devel' into fix-link-and-recommended-scss

italojs committed:
Merge pull request #13477 from meteor/fix-link-and-recommended-scss

docs: Fix links and add leonardoventurini:scss as recommended

nachocodoner committed:
Merge branch 'devel' into cache-windows-ci-check

nachocodoner committed:
re-run checks

nachocodoner committed:
re-run checks

nachocodoner committed:
clean

nachocodoner committed:
disable submodule update failure check in install.ps1

nachocodoner committed:
temp disable clean to verify cache speed

nachocodoner committed:
simplify cache paths in Windows self-test workflow

nachocodoner committed:
simplify cache paths in Windows self-test workflow

Grubba27 committed:
FEATURE: Remove `import React from 'react';` from skels

With our SWC changes, we can both remove this import line and add an example of swc configuration

nachocodoner committed:
skip temporary --get-ready

Grubba27 committed:
Merge pull request #14041 from meteor/feature/remove-react-import

FEATURE: Remove `import React from 'react';` from skels

nachocodoner committed:
refactor CI install script to adapt for CI environment

nachocodoner committed:
update workflow name to 'Windows Selftest'

nachocodoner committed:
refine Windows self-test workflow configuration

nachocodoner committed:
rename workflow file and update path references

nachocodoner committed:
fix path for windows-selftest.yml in workflow triggers

nachocodoner committed:
update cache key in Windows CI workflow

nachocodoner committed:
refactor dependency installation steps in Windows CI workflow

nachocodoner committed:
Merge pull request #14032 from meteor/cache-windows-ci-check

Windows CI check optimizations

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
remove wildcard branch trigger from Windows CI workflow

nachocodoner committed:
Merge branch 'devel' into release-3.4

nachocodoner committed:
re-run checks

italojs committed:
Merge branch 'release-3.4.1' into fix/test-in-console

italojs committed:
Merge pull request #13000 from meteor/fix/test-in-console

Fix bugs with test-in-console

nachocodoner committed:
simplify cache key by removing package-lock.json hash

italojs committed:
Merge branch 'devel' into patch-1

italojs committed:
Merge pull request #12909 from graemian/patch-1

Add instructions for enabling back gesture on iOS

italojs committed:
merge branch

nachocodoner committed:
Merge branch 'devel' into release-3.4

italojs committed:
Merge branch 'release-3.4' into release-3.4.1

italojs committed:
test: make typescript template test async and await s.init()

italojs committed:
Merge branch 'release-3.4.1' into fix/minimongo-remove-in

italojs commented:
@ggazzo do you have plans to fix this PR?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/meteor/meteor?pullRequest=13440) <br/>All committers have signed the CLA.
